packages/lint/src/types.ts

```diff
@@ -0,0 +1,51 @@
+import { Program, SemanticNodeListener } from "@cadl-lang/compiler";
+
+export interface RegisterRuleOptions {
+  enable?: boolean;
+}
+
+export interface Linter {
+  /**
```

Comment:
We recently have some requests to provide a report of 'positive' linting rule executions.  I think this will likely add to this interface, and we can design that as part of the associated design issue, but I wonder if listing the rule configuration would be a good addition here (i.e. rule-name: enabled/disabled)

Reply:
one of the idea with this is it is not built-in the compiler so would be more flexible to iterate over but if we see it being used a lot I think it would be good to consider a built-in integration in the compiler flow. With this is could show those additional informations.

Otherwise right now we could use the tracing engine to log all the rules that will be running.